### PR TITLE
feat(ocsf): add OCSF format as JSON output for AWS, Azure and GCP. Hello Amazon Security Lake!

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add OCSF JSON as an output mode.
https://schema.ocsf.io/1.0.0-rc.3/classes/security_finding

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by toniblyx — APPROVED** [review]

File: prowler/lib/outputs/json.py [hunk 1]
```diff
@@ -70,12 +85,147 @@ def fill_json_asff(finding_output, audit_info, finding, output_options):
     return finding_output
 
 
+def fill_json_ocsf(
```
Comment (jfagoagas) [inline prowler/lib/outputs/json.py:88]: Could you please move this function to another file?

File: prowler/lib/outputs/models.py [hunk 2]
```diff
@@ -228,6 +229,18 @@ def unroll_dict(dict: dict):
     return unrolled_items
 
 
+def unroll_dict_to_list(dict: dict):
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:232]: Could you please include tests for this function? And move it to another file to be able to use from another places.

File: prowler/lib/outputs/models.py [hunk 3]
```diff
@@ -621,3 +634,115 @@ class Check_Output_JSON_ASFF(BaseModel):
     Resources: List[Resource] = None
     Compliance: Compliance = None
     Remediation: dict = None
+
+
+# JSON OCSF
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:639]: Please, move the OCSF specification model to another file.

File: prowler/lib/outputs/models.py [hunk 4]
```diff
@@ -621,3 +634,115 @@ class Check_Output_JSON_ASFF(BaseModel):
...
+
+
+class Metadata(BaseModel):
+    original_time: str
+    profiles: List[str]
+    product: Product
+    version: str = "1.0.0-rc.3"
+
+
+class Check_Output_JSON_OCSF(BaseModel):
```
Comment (jfagoagas) [inline prowler/lib/outputs/models.py:715]: Why all the attributes have a default value? With that Pydantic makes no sense.

File: prowler/lib/outputs/json.py [hunk 5]
```diff
@@ -70,12 +85,147 @@ def fill_json_asff(finding_output, audit_info, finding, output_options):
...
+    finding_output: Check_Output_JSON_OCSF, audit_info, finding, output_options
+):
+    resource_region = ""
+    resource_name = ""
+    resource_uid = ""
+    finding_uid = ""
+    resource_labels = finding.resource_tags if finding.resource_tags else []
+    if finding.status == "PASS":
```
Comment (jfagoagas) [inline prowler/lib/outputs/json.py:96]: Also you should split this function into several ones to fill the model, it's hard to follow it.

File: prowler/lib/outputs/outputs.py [hunk 6]
```diff
@@ -161,6 +163,19 @@ def report(check_findings, output_options, audit_info):
                             )
                             file_descriptors["json"].write(",")
 
+                        if "json-ocsf" in file_descriptors:
+                            finding_output = Check_Output_JSON_OCSF()
```
Comment (jfagoagas) [inline prowler/lib/outputs/outputs.py:167]: Why this? And why the fill doesn't return anything?
